src/Aspire.Dashboard/Components/Pages/Resources.razor.cs

```diff
@@ -35,43 +38,58 @@ public partial class Resources : ComponentBase, IAsyncDisposable
     public required BrowserTimeProvider TimeProvider { get; init; }
     [Inject]
     public required IJSRuntime JS { get; init; }
+    [Inject]
+    public required ProtectedSessionStorage SessionStorage { get; init; }
+
+    [CascadingParameter]
+    public required ViewportInformation ViewportInformation { get; init; }
+
+    [Parameter]
+    [SupplyParameterFromQuery]
+    public string? Filter { get; set; }
```

Comment:
Text fields like this shouldn't go in the query string. Every character change updates the query string. This ruins the back button, because you have to back through each character change.

Reply:
The best option I can then think of to avoid it being lost on layout change is to inject a scoped service that contains this field. And then to clear the filter when leaving the page.